### PR TITLE
Issue 11 method body object

### PR DESCRIPTION
**Pull request by 123a6bcw**

### CODE REVIEW

File: src/test/java/core/ast/FieldAccessTest.java [hunk 1]
```diff
@@ -0,0 +1,539 @@
+package core.ast;
+
+import com.intellij.openapi.progress.util.ProgressIndicatorBase;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiField;
```
Comment (onewhl) [inline src/test/java/core/ast/FieldAccessTest.java]: There are a lot of unused imports.

File: src/test/java/core/ast/FieldAccessTest.java [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.WriteAbortedException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.junit.Assert.assertThat;
+
+public class FieldAccessTest extends LightCodeInsightFixtureTestCase {
```
Comment (onewhl) [inline src/test/java/core/ast/FieldAccessTest.java]: `LightCodeInsightFixtureTestCase` is deprecated.

File: src/test/java/core/ast/FieldAccessTest.java [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+    private File createTempFile(@NotNull String filename) {
+        try {
+            return File.createTempFile(filename, ".java");
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    private void writePrefix(@NotNull FileWriter writer, @NotNull File file) throws IOException {
```
Comment (onewhl) [inline src/test/java/core/ast/FieldAccessTest.java]: Please, merge master into this branch and run tests

File: src/test/java/core/ast/FieldAccessTest.java [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+
+        new ASTReader(projectInfo, new ProgressIndicatorBase());
+        SystemObject systemObject = ASTReader.getSystemObject();
+        MySystem mySystem = new MySystem(systemObject, true);
+        MyClass myClass = mySystem.getClassIterator().next();
+
+        MyAttribute testField = myClass.getAttributeList().get(fieldNumber);
+        List<String> entitySet = new ArrayList<>(testField.getFullEntitySet());
+
+        String fieldName = "FIELD";
```
[the remark this answers is not in the document]
Comment (123a6bcw) [reply src/test/java/core/ast/FieldAccessTest.java]: There is `fieldNumber = 2` in the commented tests.

File: src/test/java/core/ast/FieldAccessTest.java [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+
+    public void testReturnComplexStatementTest() {
+        String methodCode = "int returnComplexStatementTest() {\n" +
+                "        return this.FIELD.toString().hashCode();\n" +
+                "    }";
+
+        testMethod(methodCode, 0);
+    }
+
+    public void testSynzhronizedSyncTest() {
```
[the remark this answers is not in the document]
Comment (123a6bcw) [reply src/test/java/core/ast/FieldAccessTest.java]: Fixed

File: src/test/java/core/ast/FieldAccessTest.java [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+                "        int i = 100;\n" +
+                "        switch (i) {\n" +
+                "            case 6:\n" +
+                "                break;\n" +
+                "            case SWITCH_CASE_TEST:\n" +
+                "                break;\n" +
+                "        }\n" +
+                "    }";
+
+        //TODO PROBABLY AN INTENTED BEHAVIOUR: TEST ORIGINAL PLUGIN +
```
[the remark this answers is not in the document]
Comment (123a6bcw) [reply src/test/java/core/ast/FieldAccessTest.java]: Fixed

File: src/main/java/org/jetbrains/research/intellijdeodorant/core/ast/util/ExpressionExtractor.java [hunk 7]
```diff
@@ -444,6 +444,8 @@
             if (instanceChecker.instanceOf(arrayInitializer))
                 expressionList.add(arrayInitializer);
         } else if (expression instanceof PsiReferenceExpression) {
+            //TODO plugin does not find field accesses such as FIELD.toString()
```
[the remark this answers is not in the document]
Comment (123a6bcw) [reply src/main/java/org/jetbrains/research/intellijdeodorant/core/ast/util/ExpressionExtractor.java]: It is not, now it works. 
`this.field` still does not work.

File: src/test/java/core/ast/FieldAccessTest.java [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+        List<String> entitySet = new ArrayList<>(testField.getFullEntitySet());
+
+        String fieldName = "FIELD";
+        if (fieldNumber == 1) {
+            fieldName = "extraField";
+        } else if (fieldNumber == 2) {
+            fieldName = "SWITCH_CASE_TEST";
+        }
+
+        assertTrue(fieldName + "'s entity set does not contain given method.", entitySet.size() == 2);
```
[the remark this answers is not in the document]
Comment (123a6bcw) [reply src/test/java/core/ast/FieldAccessTest.java]: fixed

File: src/test/java/core/ast/FieldAccessTest.java [hunk 9]
```diff
@@ -0,0 +1,517 @@
...
+import org.jetbrains.research.intellijdeodorant.core.distance.MyClass;
+import org.jetbrains.research.intellijdeodorant.core.distance.MySystem;
+import org.jetbrains.research.intellijdeodorant.core.distance.ProjectInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FieldAccessTest extends LightJavaCodeInsightFixtureTestCase {
+    private String getPrefix() {
+        String className = "TestFieldAccess";
```
[the remark this answers is not in the document]
Comment (123a6bcw) [reply src/test/java/core/ast/FieldAccessTest.java]: fixed

File: src/main/java/org/jetbrains/research/intellijdeodorant/core/ast/util/ExpressionExtractor.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (123a6bcw) [reply src/main/java/org/jetbrains/research/intellijdeodorant/core/ast/util/ExpressionExtractor.java]: Sorry, it does work, I had a wrong test.

**Pull request merged**